hooks/tk-flame_actions.py

```diff
@@ -0,0 +1,383 @@
+
+        if "load_batch" in actions:
+            action_instances.append({"name": "load_batch",
+                                     "params": None,
+                                     "caption": "Create a Batch Group",
+                                     "description": "This will create a Batch Group using the media found inside the "
+                                                    "folder"})
+
+        return action_instances
+
+    def execute_multiple_actions(self, actions):
+        """
+        Executes the specified action on a list of items.
+
+        The default implementation dispatches each item from ``actions`` to
+        the ``execute_action`` method.
+
+        The ``actions`` is a list of dictionaries holding all the actions to execute.
+        Each entry will have the following values:
+
+            name: Name of the action to execute
+            sg_publish_data: Publish information coming from Shotgun
+            params: Parameters passed down from the generate_actions hook.
+
+        .. note::
+            This is the default entry point for the hook. It reuses the ``execute_action``
+            method for backward compatibility with hooks written for the previous
+            version of the loader.
+
+        .. note::
+            The hook will stop applying the actions on the selection if an error
+            is raised midway through.
+
+        :param list actions: Action dictionaries.
+        """
+        for single_action in actions:
+            name = single_action["name"]
+            sg_publish_data = single_action["sg_publish_data"]
+            params = single_action["params"]
+            self.execute_action(name, params, sg_publish_data)
+
+    def execute_action(self, name, params, sg_publish_data):
+        """
+        Execute a given action. The data sent to this be method will
+        represent one of the actions enumerated by the generate_actions method.
+
+        :param name: Action name string representing one of the items returned by generate_actions.
+        :param params: Params data, as specified by generate_actions.
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :returns: No return value expected.
+        """
+        app = self.parent
+        app.log_debug("Execute action called for action %s. "
+                      "Parameters: %s. Publish Data: %s" % (name, params, sg_publish_data))
+
+        try:
+            if name == "load_clip":
+                self._import_clip(sg_publish_data)
+
+            if name == "load_setup":
+                self._import_batch_file(sg_publish_data)
+
+            if name == "load_batch":
+                self._import_batch_group(sg_publish_data)
+        except FlameActionError as error:
+            from PySide import QtGui, QtCore
+            if QtCore.QCoreApplication.instance():
```

Comment:
Could you elaborate why this check is required? Why would the Flame loader hook be run without a QApplication ready? At the very least you should check for  QtCore.QApplication.instance(), since you want to show a UI popup.